hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieStorageType.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.uber.hoodie.common.model;
+
+public enum HoodieStorageType {
```

Comment:
Yeah I agree this should be FileFormat. 
FileFormat - Parquet (read_optimized_file_format), Avro (append_optimized_file_format)
StorageType - copy_on_write and append_and_compact (new name)
TableType - read_optimized and realtime. Depending on the StorageType - supported table types will be registered to hive. 

For this PR, I am changing HoodieStorageType to HoodieFileFormat, and we will continue discussion on StorageType and TableType and these changes can be part of the merge on read code.